src/mytnt.c

```diff
@@ -145,8 +145,9 @@ int mytnt_stmt_execute(MYTNT_STMT *stmt) {
 	}
 
 	if (reply->error) {
-		stmt->mytnt->error_info = (char *) malloc(1024 * sizeof(char));
-		strcpy(stmt->mytnt->error_info, reply->error);
+		int error_len = reply->error_end - reply->error;
+		stmt->mytnt->error_info = (char *)calloc(error_len + 1, sizeof(char));
```

Comment:
Nits:

1. `size_t` would fit better than `int` here.
2. We don't have to cast `void *` to a certain type like `char *` in C (unlike C++).